airdrop/project/contracts/distributor-contract/src/interface.sw

```diff
@@ -13,58 +13,75 @@ abi AirdropDistributor {
     ///
     /// # Arguments
     ///
-    /// * `amount` - The quantity of an asset allotted to the user to claim.
+    /// * `amount` - The quantity of tokens allotted to the user to claim.
     /// * 'key' - The index of the leaf which will be proven on the Merkle Tree.
-    /// * `num_leaves` - The number of leaves in the Merkle Tree.
     /// * `proof` - The Merkle proof to verify the user is authorized to claim.
-    /// * `to` - The user which has been allotted a quantity of the asset.
+    /// * `to` - The user which has been allotted a quantity of the tokens.
     ///
     /// # Reverts
     ///
     /// * When the claiming period has ended.
     /// * When the `to` `Identity` has already claimed.
+    /// * When the contract's balance is less than the `amount` to claim.
     /// * When the merkle proof verification failed.
     #[storage(read, write)]
-    fn claim(amount: u64, key: u64, num_leaves: u64, proof: Vec<b256>, to: Identity);
+    fn claim(amount: u64, key: u64, proof: Vec<b256>, to: Identity);
 
-    /// Initialized the contract and starts the airdrop.
+    /// Returns any unclaimed tokens when the claiming period of the airdrop has ended.
+    ///
+    /// # Reverts
     ///
-    /// Note: The `asset` contract will need to have a `mint_to` function implemented which this
-    /// airdrop contract may call.
+    /// * When the sender is not the contract's admin.
+    /// * When the claiming period is still active.
+    /// * When there are no tokens to claim.
+    #[storage(read)]
+    fn clawback();
+
+    /// Initializes the contract and starts the airdrop.
     ///
     /// # Arguments
     ///
-    /// * `asset` - The contract which is to be distributed.
+    /// * `admin` - Thee user which has the ability to clawback any unclaimed tokens.
     /// * `claim_time` - The number fo blocks the claiming period should last.
     /// * `merkleRoot` - The root of the merkle proof used to verify claiming.
+    /// * `num_leaves` - The number of leaves in the Merkle Tree.
     ///
     /// # Reverts
     ///
-    /// * The constructor has already been called.
-    #[storage(read, write)]
-    fn constructor(asset: ContractId, claim_time: u64, merkleRoot: b256);
+    /// * When the constructor has already been called.
+    /// * When no tokens are sent to the airdrop contract.
+    #[payable, storage(read, write)]
+    fn constructor(admin: Identity, claim_time: u64, merkleRoot: b256, num_leaves: u64);
```

Comment:
```suggestion
    fn constructor(admin: Identity, claim_time: u64, merkle_root: b256, num_leaves: u64);
```

Interface above, code below.
Does this cause a namespace collision with the getter below?